### ISSUE TITLE
[pruner] Redesign exit-code checking

### ISSUE THREAD

**Issue by robotrobo** [p4gauntlet/toz3#18]
We currently use the given compiler binary to check the corresponding exit code/err_string for the bug. Instead, take in a bash script from the user which does the necessary checks, maybe something like an `interestingness-test` from creduce.

**Comment by fruffy** [comment]
Yes, good idea. Trying to check for properties in the C++ code is too cumbersome. We should be able to ingest a script with arguments to the file. Then we can handle arbitrary tests.
